x/gamm/keeper/taker_fee.go

```diff
@@ -72,6 +70,39 @@ func (k Keeper) sendToTxFees(ctx sdk.Context, sender sdk.AccAddress, takerFeeCoi
 	return nil
 }
 
+// While charging taker fee, we reward the owner of the rollapp involved in swap. In that case,
+// the owner is called the beneficiary. The following cases are possible:
+//
+//	 No | In Denom    | Out Denom   | Result
+//	----|-------------|-------------|------------------------------
+//	 1  | RollApp     | RollApp     | Reward the IN RollApp owner
+//	 2  | RollApp     | Non-RollApp | Reward the IN RollApp owner
+//	 3  | Non-RollApp | RollApp     | Reward the OUT RollApp owner
+//	 4  | Non-RollApp | Non-RollApp | No one is rewarded
+//
+// Return nil beneficiary address if no one is rewarded: case (4) or error.
+func (k Keeper) getTakerFeeBeneficiary(ctx sdk.Context, inDenom, outDenom string) *sdk.AccAddress {
+	if k.rollappKeeper == nil {
```

Comment:
not needed to check here. keeper initialization assumed as valid
can add validation to `NewKeeper(..)`

Reply:
this keeper is set to `nil` by default in osmo repo since there's no rollapp keeper here (we do not want to import it to avoid the dependency cycle)

Reply:
added a comment